tests/datasource/test_new_datasource.py

```diff
@@ -235,13 +235,13 @@ def test_basic_pandas_datasource_v013_self_check(basic_pandas_datasource_v013):
 
 
 def test_basic_spark_datasource_self_check(basic_spark_datasource):
-    report = basic_spark_datasource.self_check()
-    report["execution_engine"]["spark_config"].pop("spark.app.id", None)
-    report["execution_engine"]["spark_config"].pop("spark.driver.host", None)
-    report["execution_engine"]["spark_config"].pop("spark.driver.port", None)
-    report["execution_engine"]["spark_config"].pop("spark.submit.pyFiles", None)
-    report["execution_engine"]["spark_config"].pop("spark.app.startTime", None)
-    report["execution_engine"]["spark_config"].pop("spark.sql.warehouse.dir", None)
+    report: dict = basic_spark_datasource.self_check()
+
+    # The structure of this config is dynamic based on PySpark version;
+    # we deem asserting it's presence sufficient for purposes of this test
```

Comment:
@anthonyburdi "it's" -> "its" :)

Reply:
will follow up with a subsequent PR or I owe you chocolate